src/app/shared/filter/filter.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs/Subscription';
+import { selectFilter } from '../../ngrx/reducers';
+import { DateRangeModel, EpisodeModel, FilterModel, IntervalModel } from '../../ngrx/model';
+import { CastleFilterAction } from '../../ngrx/actions';
+
+@Component({
+  selector: 'metrics-filter',
+  template: `
+    <metrics-date-range [filter]="filter" (dateRangeChange)="onDateRangeChange($event)"></metrics-date-range>
+    <hr>
+    <metrics-interval [filter]="filter" (intervalChange)="onIntervalChange($event)"></metrics-interval>
+    <metrics-episodes [filter]="filter" (episodesChange)="onEpisodesChange($event)"></metrics-episodes>
+    <div class="buttons">
+      <metrics-prev-date-range></metrics-prev-date-range>
+      <button (click)="onApply()">Apply</button>
```

Comment:
Could be nice to disable this when filter hasn't been changed -- visual confirmation to user that what they're looking at matches what's in controls